src/toolchain.ml

```diff
@@ -137,9 +137,11 @@ module Package_manager = struct
     | Custom _ -> "Custom OCaml"
 end
 
-type resources = Package_manager.t
+type t = { package_manager : Package_manager.t }
```

Comment:
The new name is good, but we don't need this extra record.

Reply:
I wanted to avoid type aliasing for reasons:
- to avoid lower-quality type-on-hover as is the case with `Cmd` module `type stderr = string` and `type stdout = string`, which ends up making all strings `stderr` (this is less acute here, but I'd love to keep it as it is
- having `package_manager` as a part of `Toolchain.t` makes it (1) easily extensible and (2) shows that a `package_manager` is only a _part_ of what a toolchain is

Please let's keep it :-)

Reply:
Okay, you can keep because of the hovers.

I don't think it's worth it to complicate current code to make it easier to *possibly* add more code in the future. So making things extensible should have a very concrete extension in mind. OCaml already makes it easy to refactor, there's no need to leave unused hooks for extension everywhere.

Reply:
We might want to remove the distinction between the two types entirely. I think there's a historical reason why there are two different types, but currently I feel like it complicates the code by requiring conversions between them.

Reply:
> We might want to remove the distinction between the two types entirely. I think there's a historical reason why there are two different types, but currently I feel like it complicates the code by requiring conversions between them.

I thought about it but I'm yet to come up with proper abstractions, and I didn't want to change _too_ much.

Reply:
In that case, lets revert the Toolchain changes and handle it in a separate PR.

Reply:
I'd still like this change to be reverted, since the `Toolchain` module will be revisited in another PR.

Reply:
I provided specific arguments why I wouldn't like to have a type alias with which Rudi, I think, is okay. 

I don't want to revert and redo things later just for the sake of having another PR. If you want to do that, go ahead.